src/main/java/org/zalando/nakadi/service/WebResult.java

```diff
@@ -0,0 +1,21 @@
+package org.zalando.nakadi.service;
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.context.request.NativeWebRequest;
+import org.zalando.problem.spring.web.advice.Responses;
+
+import java.util.function.Supplier;
+
+import static org.springframework.http.HttpStatus.OK;
+import static org.springframework.http.ResponseEntity.status;
+
+public class WebResult {
+
+    public static ResponseEntity<?> wrap(final Supplier<Result> supplier, final NativeWebRequest request) {
```

Comment:
@v-stepanov agree, i also do not like it much but for now I just wrap the same exceptions in one method it reduces some amount of duplicated code. I think a right way is to use RuntimeException and handle it in @ControllerAdvice. It will give us less duplicated code and have one place where we handle all exceptions.